demos/showcase/src/app/app.component.html

```diff
@@ -1,13 +1,25 @@
-<pxb-drawer-layout [variant]="getVariant()" (backdropClick)="closeDrawer()">
+<pxb-drawer-layout
+    [variant]="getVariant()"
+    (backdropClick)="closeDrawer()"
+    [dir]="isRtl ? 'rtl' : 'ltr'"
+    [class.rtl]="isRtl"
```

Comment:
Why do we need the dir attribute and the class? Also, shouldn't we be putting this on the `<body>` instead of the DrawerLayout?

Reply:
You're right about the rtl class, it's not required.

Even though I'm not putting the dir attribute on the body, the overlay component is still being updated in the dom to have its `dir="rtl"`.   As I recall, the theme was the item that was causing issues with the ng overlay container, since when we applied the theme on the app root, the overlay wasn't being styled.  

![image](https://user-images.githubusercontent.com/6538289/88545032-6c074700-cfe8-11ea-87fb-6244b6c5dcfa.png)

Seems like overlay dir functionality is built into angular _somewhere_.

Reply:
I'm just curious why we are putting the dir on the DrawerLayout instead of on the body if our instructions for doing RTL languages say to do it on body.

Reply:
Very strange that the overlay is picking it up even though none of its parents have RTL set...

Reply:
DrawerLayout uses the Direction service (provided by angular) to detect `dir` attributes in its parent DOM tree.  Ive been testing this out for about an hour now; this directive does not detect changes unless you actually use the `dir` directive.

This means if you use javascript to set the dir attribute on the body, or even the app-root, it is ignored by the Direction service and the Drawer doesn't flip to the other side of the screen.

The DrawerLayout component is the only component that actually relies on the DirectionService.  All of our other bidirectionality styles are strictly through css, so they work as long as a parent element in the DOM tree has the `dir` attribute set.

This issue is also present in our storybook app; to test out RTL, I'm using JS to update the dir attribute.  The `within a DrawerLayout story` is broken because the Direction directive `dir` is not used.

Our instructions should be updated for angular to use the `dir` directive instead of setting the `dir` attribute through js.

So this works: 
<div dir='rtl'> .... </div>

But this won't work:
document.body.dir = 'rtl';

Reply:
> So this works:
> `<div dir='rtl'> .... </div>`
> 
> But this won't work:
> `document.body.dir = 'rtl'; `

🤯  whaaat? These are not the same thing?

Reply:
These are not the same thing, but only because the angular service `Direction` (which DrawerLayout uses for inline styles) explicitly needs the `dir` directive to be used within a template.

Reply:
Ok, so if we're putting it on DrawerLayout, but somebody else has a project that needs RTL, and they're not using DrawerLayout, they would just have to put in on whatever their top-level item is in their app-root?

Reply:
Yes that's correct, and unfortunately it cannot be placed directly on their app-root either, like you said it has to be the first element within their app-root.

So this won't work: 
```
    <body class="pxb-blue mat-typography mat-app-background">
        <app-root dir="'rtl'"></app-root>
    </body>
```